src/Microsoft.Health.Dicom.Client/DicomWebClient.cs

```diff
@@ -157,34 +157,35 @@ public async Task<DicomWebResponse<IReadOnlyList<DicomDataset>>> RetrieveMetadat
             string studyInstanceUid = null,
             CancellationToken cancellationToken = default)
         {
-            var postContent = new List<byte[]>();
+            var postContent = new List<Stream>();
 
             foreach (DicomFile dicomFile in dicomFiles)
             {
-                await using (var stream = GetMemoryStream())
+                var stream = GetMemoryStream();
+                await dicomFile.SaveAsync(stream);
+                stream.Seek(0, SeekOrigin.Begin);
+                postContent.Add(stream);
+            }
+
+            try
```

Comment:
try should be above or right after GetMemoryStream(), so that even if dicomFile.SaveAsync failes the finally is executed.